Mathlib/RingTheory/IntegrallyClosed.lean

```diff
@@ -11,8 +11,8 @@ import Mathlib.RingTheory.Localization.Integral
 /-!
 # Integrally closed rings
 
-An integrally closed domain `R` contains all the elements of `Frac(R)` that are
-integral over `R`. A special case of integrally closed domains are the Dedekind domains.
+An integrally closed ring `R` contains all the elements of `Frac(R)` that are
+integral over `R`. A special case of integrally closed rings are the Dedekind domains.
```

Comment:
In the Stacks Project, integral closedness is defined for any Algebra, not necessarily the FractionRing over the base ring: https://stacks.math.columbia.edu/tag/00GP Maybe we should change IsIntegrallyClosed to take the algebra as an argument and show that it's equivalent to [integralClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/IntegralClosure.html#integralClosure) being \top?

The current definition is called [Normal](https://stacks.math.columbia.edu/tag/0309) in Stacks, with the IsDomain condition. A [normal *ring*](https://stacks.math.columbia.edu/tag/00GV) means that localizations at all prime ideals are normal (domains). This [implies](https://stacks.math.columbia.edu/tag/034M) integral closedness in its FractionRing, but is equivalent to it only [under some conditions](https://stacks.math.columbia.edu/tag/030C) (reduced + finitely many minimal primes), in which case it's also equivalent to being a finite product of normal domains. Whether we use Normal or IntegrallyClosed (in FractionRing) in the definition of IsDedekindRing, it looks like we'd need to add these conditions if we want exactly the products of Dedekind domains.

Reply:
I've now found [a paper](https://www.ams.org/journals/proc/1989-105-04/S0002-9939-1989-0942636-0/S0002-9939-1989-0942636-0.pdf) in which integrally-closedness is defined for a general commutative ring to mean it's integrally closed in its FractionRing. Therefore I think this PR is reasonable.

Reply:
Just because such a paper exists doesn't mean the definition is right, however? I think it's still fine to go ahead with this PR, but maybe it'd be nice to leave a note about all this for future work to maybe change this.

I do worry that actually this future definition that is defined for any algebra will cause issues with typeclass search, but I guess that can be seen in the future.

Reply:
I agree that it would be good to record most of Junyan's comment here on GH into a comment in the source. @alreadydone @Vierkantor could one of you do that please?

Reply:
Defining `IsIntegrallyClosed` with respect to any `Algebra` would probably work fine, at the cost of having to insert more parameters. How about I make a follow-up PR to this one that does that, and we can see separately how much has to be changed to do that? And whoever needs normal domains and rings can then build on that result :)

Reply:
Done: #7116

Reply:
Thanks!